### ISSUE TITLE
Error when uncaching using a key with a pattern

### ISSUE THREAD

**Issue by jeky1950** [yxx4c/prisma-extension-redis#3]
Node Version: v18.17.1
NPM Version: 10.2.0
Prisma Client: 5.17.0
@yxx4c/prisma-redis-extension: 0.0.4
ioredis: 5.4.1
Redis Server: 7.4.0

Caching is working and is done using the following code:

```
try {
      const product = await prisma.product.findMany({
            where: {
                name: {
                      contains: search
                }
            },
            take: 5,
            cache: { ttl: 60 * 5, key: getCacheKey([ { prisma: 'Product' }, { search } ]) }
        });
} catch (error) {
    console.error(error)
}
```

When I use the following code to uncache it actually deletes the record from my Redis Server but the endpoint breaks:

```
try {
       const updateProduct = await prisma.product.update({
                    where: { id },
                    data: { quantity: { decrement: quantity } },
                    uncache: {
                        uncacheKeys: [ getCacheKeyPattern([ { prisma: 'Product' }, { glob: '*' } ]) ],
                        hasPattern: true
                    }
                });
} catch (error) {
    console.error(error)
}
```

And I get this error in the terminal

```
 ERROR  ERR wrong number of arguments for 'unlink' command

  at parseError (node_modules\redis-parser\lib\parser.js:179:12)
  at parseType (node_modules\redis-parser\lib\parser.js:302:14)
```

**Comment by jeky1950** [comment]
I have done more testing and the error doesnt occur when I set hasPattern to false. Unfortunately I need to use patterns.

**Comment by jeky1950** [comment]
After a few hours digging I was able to create a work around I found [here](https://stackoverflow.com/a/36006360/19910602) by using **del** instead of **unlink** on the redis pipeline method inside your **unlinkPattern** function like so:

```
const unlinkPatterns = (pattern) => new Promise((resolve) => {
  const stream = redis.scanStream({
    match: pattern
  });

  stream.on('data', function (keys) {
    // `keys` is an array of strings representing key names
    if (keys.length) {
      var pipeline = redis.pipeline();
      keys.forEach(function (key) {
        pipeline.del(key);
      });
      pipeline.exec();
    }
  });

  stream.on('end', function () {
    resolve(true);
  });
});
```

And then called it after the prisma call like so:

```
const updateProduct = await prisma.product.update(...);

await unlinkPatterns(getCacheKeyPattern([ { prisma: 'Product' }, { company_id }, { glob: '*' }]))
```

**Comment by yxx4c** [comment]
Hi @jeky1950 

Thank you for bringing this to my attention! I’ve identified the issue: it arises from passing only one key to be uncached, which is a pattern. The code was originally designed to uncache keys and was later updated to handle patterns. The `hasPattern` toggle in the config allows patterns to be processed separately, while non-patterns are handled normally.

The error you encountered, `ERR wrong number of arguments for 'unlink' command`, occurred because the non-pattern unlink operation attempted to unlink an empty array.

I will push an update to resolve this issue as soon as possible. In the meantime, you can use an invalid key as a temporary workaround for uncaching. Here’s an example:

```ts
try {
    const updateProduct = await prisma.product.update({
        where: { id },
        data: { quantity: { decrement: quantity } },
        uncache: {
            uncacheKeys: [ 
                getCacheKeyPattern([{ prisma: 'Product' }, { glob: '*' }]),
                "invalid_key_as_temporary_workaround"
            ],
            hasPattern: true
        }
    });
} catch (error) {
    console.error(error);
}
```

Let me know if you have any further questions!

**Comment by jeky1950** [comment]
Hey, just tried it out and it worked

**Comment by yxx4c** [comment]
Hi @jeky1950,

I have fixed the bug and updated the package (v0.1.0). Please check it and let me know if you encounter any further issues.
